src/ott/tools/plot.py

```diff
@@ -100,13 +104,46 @@ def _scatter(self, ot: Transport):
 
   def _mapping(self, x: jnp.ndarray, y: jnp.ndarray, matrix: jnp.ndarray):
     """Compute the lines representing the mapping between the 2 point clouds."""
+    # Only plot the lines with a cost above the threshold.
     u, v = jnp.where(matrix > self._threshold)
     c = matrix[jnp.where(matrix > self._threshold)]
     xy = jnp.concatenate([x[u], y[v]], axis=-1)
+
+    # Check if we want to adjust transparency.
+    adjust_transparency = self._adjust_transparency
+
+    # We can only adjust transparency if max(c) > min(c).
+    if adjust_transparency:
+      min_matrix, max_matrix = jnp.min(c), jnp.max(c)
+      if max_matrix == min_matrix:
+        adjust_transparency = False
+
+    # Initialize the list of lines of the form (start, end, strength, alpha).
     result = []
+
+    # Loop over the lines and add them to the list.
     for i in range(xy.shape[0]):
+
+      # Line strength is coupling strength times the number of points.
       strength = jnp.max(jnp.array(matrix.shape)) * c[i]
-      result.append((xy[i, [0, 2]], xy[i, [1, 3]], strength))
+
+      # Compute the transparency.
+      if adjust_transparency:
+
+        # make alpha proportional to the coupling strength and between 0 and 1.
+        normalized_strength = (c[i] - min_matrix) / (max_matrix - min_matrix)
+        alpha = self._alpha * float(normalized_strength)
+
+        # Matplotlib's transparency is sensitive to numerical errors.
+        alpha = min(1.0, alpha)
+      else:
+        alpha = self._alpha
+
+      # Get start and end points, and append the line to the list.
```

Comment:
Would remove this comment, the line it's commenting is really simple/intent is obvious.